provision/InventoryFilters/GenericInventory.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import abc
+import StringIO
+from ansible import errors
+try:
+    from configparser import ConfigParser
+except ImportError:
+    from ConfigParser import ConfigParser
+from AWSInventory import AWSInventory
+from OpenstackInventory import OpenstackInventory
+from GCloudInventory import GCloudInventory
+from InventoryFilter import InventoryFilter
+
+
+class GenericInventory(InventoryFilter):
+    def __init__(self):
+        InventoryFilter.__init__(self)
+        self.filter_classes = {
```

Comment:
Agreed , 
Whats you call on design which is similar to as follows : 
https://github.com/apache/libcloud/blob/trunk/libcloud/compute/providers.py